doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,258 @@
+"""
+============================================
+Restore spotted cornea image with inpainting
+============================================
+
+Optical coherence tomography (OCT) is a non-invasive imaging technique used by
+ophthalmologists to take pictures of the back of a patient's eye [1]_.
+When performing OCT,
+dust may stick to the reference mirror of the equipment, causing dark spots to
+appear on the images. The problem is that these dirt spots cover areas of
+in-vivo tissue, hence hiding data of interest. Our goal here is to restore
+(reconstruct) the hidden areas based on the pixels near their boundaries.
+
+This tutorial is adapted from an application shared by Jules Scholler [2]_.
+The images were acquired by Viacheslav Mazlin (see
+:func:`skimage.data.palisades_of_vogt`).
+
+.. [1] David Turbert, reviewed by Ninel Z Gregori, MD (2023)
+       `What Is Optical Coherence Tomography?
+       <https://www.aao.org/eye-health/treatments/what-is-optical-coherence-tomography>`_,
+       American Academy of Ophthalmology.
+.. [2] Jules Scholler (2019) "Image denoising using inpainting"
+       https://www.jscholler.com/2019-02-28-remove-dots/
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import plotly.io
+import plotly.express as px
+from scipy import sparse
+
+import skimage as ski
+
+
+#####################################################################
+# The dataset we are using in this example is an image sequence (a movie!) of
+# human in-vivo tissue. Specifically, it shows the *palisades of Vogt* of a
+# given cornea sample.
+
+#####################################################################
+# Load image data
+# ===============
+
+image_seq = ski.data.palisades_of_vogt()
+
+print(f'number of dimensions: {image_seq.ndim}')
+print(f'shape: {image_seq.shape}')
+print(f'dtype: {image_seq.dtype}')
+
+#####################################################################
+# The dataset is an image stack with 60 frames (time points) and 2 spatial
+# dimensions. Let us visualize 10 frames by sampling every six time points:
+# We can see some changes in illumination.
+# We take advantage of the ``animation_frame`` parameter in
+# Plotly's ``imshow`` function. As a side note, when the
+# ``binary_string`` parameter is set to ``True``, the image is
+# represented as grayscale.
+
+fig = px.imshow(
+    image_seq[::6, :, :],
+    animation_frame=0,
+    binary_string=True,
+    labels={'animation_frame': '6-step time point'},
+    title='Sample of in-vivo human cornea'
+)
+plotly.io.show(fig)
+
+#####################################################################
+# Aggregate over time
+# ===================
+# First, we want to detect those dirt spots where the data are lost. In
```

Comment:
Look up the infamous question "is data singular or plural" :wink: (hint: it's Latin).

I believe that 'where' is clearer, because here we are interested in spatial locations... *where* the signal is lost! I would use "for which" when discussing region properties; for example, "areas for which the bounding box is square" (ok, I guess 'whose' or 'where' would work as well, and probably sound better).

Reply:
Okay, but shouldn't it then be
```suggestion
# First, we want to detect those dirt spots where the data is lost. In
```
as I believe the verb is referring to "data" which you point out is singular?

Reply:
I (indirectly) pointed out: "The word data is a plural noun so write "data are". Datum is the singular." (first search result) Nowadays, both usages are standard.

Reference: https://www.theguardian.com/news/datablog/2010/jul/16/data-plural-singular

Reply:
It wasn't clear to me from your comment which one you were using here. My impression is that singular data is the most common option by a large margin. But that may be just my particular subjective bubble and if you feel strongly about this, please ignore this thread.

Not sure what your impression is, but it feels like we tend to bikeshed quite a bit on these kind of things? :sweat_smile:

Reply:
Yes, this is definitely bikeshedding! :sweat_smile: (moving the conversation to Zulip  :wink:)